device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/TopicParser.java

```diff
@@ -36,25 +36,23 @@ protected String getStatus(int tokenIndexStatus) throws TransportException
             throw new TransportException(new IllegalArgumentException("Invalid token Index for status"));
         }
 
-        if (topicTokens.length > tokenIndexStatus)
```

Comment:
we have a check above which throws an exception if tokenIndexStatus <=0 or if it is >= topic.length. This ensures that once control reaches this line, tokenIndexStatus is always < topic.lenth.
Hence, this `if` block is unnecessary.